### PR TITLE
[Bug 1762430] Generate ExternalTaskMarkers for Airflow downstream dependencies

### PR DESCRIPTION
**Pull request by scholtzan**
See https://bugzilla.mozilla.org/show_bug.cgi?id=1762430

This updates the Airflow DAG generation logic to generate `ExternalTaskMarker`s for downstream tasks. This ensures that when clearing a task for a rerun, all downstream tasks get automatically cleared as well. Newer versions of Airflow (those with a grid view) nicely group all the tasks part of a `TaskGroup` instead of showing the individual tasks.

This update also makes a change to use Airflow's `ExternalTaskSensor` instead of the custom `ExternalTaskCompletedSensor`.

The change diff looks scarily large, but the actual code changes are pretty minimal. Most changes are due to the Airflow DAGs having been regenerated.

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

File: bigquery_etl/query_scheduling/templates/airflow_dag.j2 [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # Generated via https://github.com/mozilla/bigquery-etl/blob/main/bigquery_etl/query_scheduling/generate_airflow_dags.py
 
 from airflow import DAG
-from operators.task_sensor import ExternalTaskCompletedSensor
```
Comment (scholtzan) [inline bigquery_etl/query_scheduling/templates/airflow_dag.j2:4]: Using `ExternalTaskSensor` instead. This is causing almost all of the DAGs to have some changes. If this is annoying for reviewing this, then I can also open a separate PR for this change.

**Review by whd — APPROVED** [review]
I'm approving this from a strictly CODEOWNERS/change control perspective as this change need not be subject to it

**Review by jklukas — APPROVED** [review]
The code changes here look good. 

Do we have a written plan for communicating the change and gameplan for rolling it out and testing or rolling back as needed? In particular, I think this should be coordinated with whoever is on Airflow triage in case it causes some unintended breakage or needs to be rolled back.

**Comment by scholtzan** [comment]
I was checking if there is an option to disable "Downstream Recursive" clears by default, to avoid clearing downstream tasks unnecessarily. However, there is no such setting in Airflow.

**Pull request merged**